src/QsCompiler/SyntaxProcessor/SymbolOccurrence.fs

```diff
@@ -0,0 +1,194 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+namespace Microsoft.Quantum.QsCompiler.SyntaxProcessing
+
+open System
+open System.Collections.Immutable
+open System.Diagnostics.CodeAnalysis
+open Microsoft.Quantum.QsCompiler.DataTypes
+open Microsoft.Quantum.QsCompiler.SyntaxTokens
+
+type SymbolOccurrence =
+    | Declaration of QsSymbol
+    | UsedType of QsType
+    | UsedVariable of QsSymbol
+    | UsedLiteral of QsExpression
+
+module SymbolOccurrence =
+    let rec flattenSymbol symbol =
+        match symbol.Symbol with
+        | SymbolTuple items -> Seq.collect flattenSymbol items |> Seq.toList
+        | _ -> [ symbol ]
+
+    let symbolDeclarations = flattenSymbol >> List.map Declaration
+
+    let rec inType type_ =
```

Comment:
Can we please still use the private keyword for the corresponding functions also when a .fsi is present? It is somewhat cumbersome to have to open the fsi to figure out which things are public and which things are not.

Reply:
I guess this is a matter of style. The F# style guide doesn't explicitly discourage `private` if you use a signature file, but it gets close with [this sentence](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/component-design-guidelines#hide-the-representation-of-f-union-types-in-vanilla-net-apis):

> When designing a vanilla .NET API, consider hiding the representation of a union type by using either a private declaration or a signature file.

"Either ... or ..." is exclusive, so it seems to suggest that you should use `private` in the .fs file, or omit it from an .fsi file, but not both.

My opinion is: the .fsi file *defines* the public interface for the .fs file, so I think it is a good thing that you need to open it to figure out what is public.

Reply:
The style guide here I believe is vague. As such, I have no objections keeping it as is, merely a preference as stated.